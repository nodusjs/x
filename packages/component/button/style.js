import { css } from "@nodusjs/std/dom";

export const style = (button) =>
  css`
    :host {
      width: ${button.onlyIcon ? `var(--button-width-${button.size})` : button.width};

      button {
        align-items: center;
        aspect-ratio: ${button.onlyIcon ? "1/1" : "auto"};
        background-color: var(--button-background-color-${button.color}-${button.variant});
        border: var(--button-border-${button.color}-${button.variant});
        border-radius: var(--radius-md);
        box-sizing: border-box;
        color: var(--button-color-${button.color}-${button.variant});
        cursor: pointer;
        display: inline-flex;
        font-family: var(--font-family-base);
        font-size: var(--button-font-size-${button.size});
        font-weight: var(--font-weight-semibold);
        gap: var(--button-gap-${button.size});
        height: var(--button-height-${button.size});
        justify-content: center;
        line-height: 1;
        padding: 0 ${button.onlyIcon ? 0 : "var(--spacing-lg)"};
        transition: all 0.2s ease-out;
        width: ${button.onlyIcon ? `var(--button-width-${button.size})` : button.width};

        &:hover {
          background-color: var(--button-background-color-${button.color}-${button.variant}_hover);
          color: var(--button-color-${button.color}-${button.variant}_hover);
        }

        &:focus-visible {
          box-shadow: var(--button-box-shadow-${button.color});
          outline: 0;
        }
      }
    }

    :host(:state(disabled)) {
      button {
        background-color: var(--bg-disabled);
        border: 1px solid var(--border-disabled_subtle);
        color: var(--fg-disabled);
        cursor: auto;
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
