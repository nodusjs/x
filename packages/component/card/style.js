import { css } from "@nodusjs/std/dom";

export const style = (card) =>
  css`
    :host {
      background-color: var(--bg-primary);
      border: 1px solid var(--border-secondary);
      border-radius: var(--radius-xl);
      box-sizing: border-box;
      container-type: inline-size;
      display: inline-flex;
      height: ${card.height};
      padding: var(--spacing-${card.spacing});
      width: ${card.width};
      width: var(--width-${card.width}, ${card.width});
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
