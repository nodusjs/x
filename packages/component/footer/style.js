import { css } from "@nodusjs/std/dom";

/**
 * @function style
 *
 * @returns {CSSStyleSheet} A folha de estilo final para o rodapé.
 *
 * @description
 * Gera a folha de estilo (`CSSStyleSheet`) para o componente `x-footer`.
 *
 * Esta função define o layout principal do rodapé, estabelecendo-o como um
 * contêiner flexível de largura total com uma altura fixa e responsiva.
 * Ela também gerencia o estado de visibilidade do componente através da
 * pseudo-classe `:state(hidden)`.
 */
export const style = () =>
  css`
    :host {
      align-items: center;
      box-sizing: border-box;
      display: flex;
      height: 72px;
      width: 100%;

      @content (min-width: 768px) {
        height: 80px;
      }

      footer {
        container-type: inline-size;
        width: 100%;

        content {
          display: flex;
          gap: var(--spacing-2xl);
          justify-content: end;
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
