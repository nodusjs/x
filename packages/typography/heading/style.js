import { css } from "@nodusjs/std/dom";

/**
 * Gera o CSS específico para o `<x-heading>`, sobrescrevendo os estilos
 * base do `<x-text>` para exibir tipografia de display.
 *
 * @param {Object} text                    Instância de `Heading` (extends Text).
 * @param {('xs'|'sm'|'md'|'lg'|'xl'|'2xl')} text.size   Tamanho tipográfico do heading.
 * @returns {CSSStyleSheet}                Folha de estilo contendo apenas as regras que ajustam font-size e line-height.
 *
 * @description
 * Esta função injeta dinamicamente as variáveis CSS de display baseadas nos
 * tokens `--font-size-display-{size}` e `--line-height-display-{size}`, permitindo
 * que o `<x-heading>` utilize níveis de título (h1, h2, etc.) com escala de
 * tipografia adequada.
 *
 * @example
 * import Heading from "./heading.js";
 * import { style as headingStyle } from "./style.js";
 *
 * const h = new Heading();
 * h.size = "lg";
 *
 * const sheet = headingStyle(h);
 * // sheet.cssRules[0].cssText será algo como:
 * // ":host { font-size: var(--font-size-display-lg); line-height: var(--line-height-display-lg); }"
 */
export const style = (heading) =>
  css`
    :host {
      ${heading.as} {
        font-size: var(--font-size-display-${heading.size});
        line-height: var(--line-height-display-${heading.size});
      }
    }
  `;
