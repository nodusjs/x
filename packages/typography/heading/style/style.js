import { css } from "@nodusjs/std/dom";

export const style = (text) =>
	css`
    :host {
      font-size: var(--font-size-display-${text.size});
      line-height: var(--line-height-display-${text.size});
    }
  `;
