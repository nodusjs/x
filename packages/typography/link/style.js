import { css } from "@nodusjs/std/dom";

export const style = (text) =>
  css`
    :host {
      box-sizing: border-box;
      display: inline-flex;

      a {
        color: var(--text-brand-secondary);
        cursor: pointer;
        display: inline-flex;
        font-family: var(--font-family-base);
        font-size: var(--font-size-text-${text.size});
        font-weight: var(--font-weight-semibold);
        line-height: var(--line-height-text-${text.size});
        margin: 0;
        max-width: var(--paragraph-max-width);
        text-wrap: ${text.wrap};
        transition: all 0.2s ease-out;
        width: auto;

        &:hover {
          color: var(--text-brand-secondary_hover);
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
