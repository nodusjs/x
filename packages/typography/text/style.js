import { css } from "@nodusjs/std/dom";

export const style = (text) =>
  css`
    :host {
      box-sizing: border-box;
      display: inline-flex;
      max-width: var(--paragraph-max-width);

      ${text.as} {
        color: var(--text-${text.color}${text.onBrand ? "_on-brand" : ""});
        font-family: var(--font-family-base);
        font-size: var(--font-size-text-${text.size});
        font-weight: var(--font-weight-${text.weight});
        line-height: var(--line-height-text-${text.size});
        margin: 0;
        text-align: ${text.align};
        text-wrap: ${text.wrap};
        transition: all 0.2s ease-out;
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
