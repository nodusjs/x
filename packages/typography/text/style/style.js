import { css } from "@nodusjs/std/dom";

export const style = (text) =>
  css`
    :host {
      ${text.as} {
        color: var(--text-${text.color}${text.onBrand ? "_on-brand" : ""});
        display: inline-flex;
        font-family: var(--font-family-base);
        font-size: var(--font-size-text-${text.size});
        font-weight: var(--font-weight-${text.weight});
        line-height: var(--line-height-text-${text.size});
        max-width: var(--paragraph-max-width);
        text-align: ${text.align};
        text-wrap: ${text.wrap};
        transition: all 0.2s ease-out;
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
