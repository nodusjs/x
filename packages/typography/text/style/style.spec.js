import { describe, it, expect } from 'vitest'
import { style } from './style.js'

describe('style', () => {
  it('gera o CSS esperado com valores padrÃ£o', () => {
    const text = {
      as:      'p',
      color:   'primary',
      onBrand: false,
      size:    'md',
      weight:  'medium',
      align:   'left',
      wrap:    'wrap'
    }

    const { cssRules: [{ cssText }]} = style(text)
    expect(cssText).toMatchSnapshot()
  })

  it('gera o CSS esperado com valores customizados e on-brand', () => {
    const text = {
      as:      'p',
      color:   'secondary',
      onBrand: true,
      size:    'xl',
      weight:  'bold',
      align:   'center',
      wrap:    'no-wrap'
    }

    const { cssRules: [{ cssText }]} = style(text)
    expect(cssText).toMatchSnapshot()
  })
})
