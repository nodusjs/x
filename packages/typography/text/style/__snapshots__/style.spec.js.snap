// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`style > gera o CSS esperado com valores customizados e on-brand 1`] = `":host { color: var(--text-secondary_on-brand); display: inline-flex; font-family: var(--font-family-base); font-size: var(--font-size-text-xl); font-weight: var(--font-weight-bold); line-height: var(--line-height-text-xl); text-align: center; text-wrap: no-wrap; width: var(--paragraph-max-width); }"`;

exports[`style > gera o CSS esperado com valores padr√£o 1`] = `":host { color: var(--text-primary); display: inline-flex; font-family: var(--font-family-base); font-size: var(--font-size-text-md); font-weight: var(--font-weight-medium); line-height: var(--line-height-text-md); text-align: left; text-wrap: wrap; width: var(--paragraph-max-width); }"`;
